aggregator/src/main/java/com/vmware/flowgate/aggregator/scheduler/job/AggregatorService.java

```diff
@@ -137,6 +146,85 @@ public void mergeServerMapping() {
       }
    }
 
+   public void aggregateAndCleanPDUFromPowerIQ() {
+      restClient.setServiceKey(serviceKeyConfig.getServiceKey());
+      FacilitySoftwareConfig[] powerIQs = restClient.getFacilitySoftwareByType(SoftwareType.PowerIQ).getBody();
+      if(powerIQs ==null || powerIQs.length==0) {
+         logger.info("No PowerIQ server find");
+         return;
+      }
+      Map<String,Asset> pdusOnlyFromPowerIQ = new HashMap<String,Asset>();
+      for(FacilitySoftwareConfig powerIQ : powerIQs) {
+         List<Asset> pdus = restClient.getAllAssetsBySourceAndType(powerIQ.getId(),AssetCategory.PDU);
+         for(Asset pdu : pdus) {
+          if(pdu.getAssetSource().split(FlowgateConstant.SPILIT_FLAG).length == 1) {
+             pdusOnlyFromPowerIQ.put(pdu.getAssetName().toLowerCase(), pdu);
+          }
+         }
+      }
+      if(pdusOnlyFromPowerIQ.isEmpty()) {
+         logger.info("No Pdu from PowerIQ server find");
+         return;
+      }
+      List<Asset> pdus = restClient.getAllAssetsByType(AssetCategory.PDU);
+      List<Asset> pdusNeedToUpdate = new ArrayList<Asset>();
+      ObjectMapper mapper = new ObjectMapper();
+      for(Asset pdu : pdus) {
+         Asset pduFromPowerIQ = pdusOnlyFromPowerIQ.get(pdu.getAssetName().toLowerCase());
+         if(pduFromPowerIQ != null && !pdu.getId().equals(pduFromPowerIQ.getId())) {
```

Comment:
I think this will not happens since getID will return the Asset ID, if two object have the same ID ,then you don;'t need to do anything.

Reply:
!pdu.getId().equals(pduFromPowerIQ.getId()  I think this check is useless. 

Since asset: pduFromPowerIQ it will only have one source which is one of the POWERIQ server. it's a new Asset.

for any Assets that left in pdus, it's ether  have two sources or it from none PowerIQ systems.  

Then the ID of the two asset will never equal.

Reply:
Done